coalib/bearlib/naming_conventions/__init__.py

```diff
@@ -130,3 +130,45 @@ def to_spacecase(string):
     return re.sub('(?<=[^\s])([A-Z])',
                   lambda match: ' ' + match.group(1),
                   string)
+
+
+def to_kebabcase(string):
+    """
+    Converts the given string to kebab-case.
+
+    >>> to_kebabcase('HelloWorld')
+    'hello-world'
+    >>> to_kebabcase('__Init__File__')
+    '--init-file--'
+    >>> to_kebabcase('')
+    ''
+    >>> to_kebabcase('already-kebab-case')
+    'already-kebab-case'
+    >>> to_kebabcase('   string  ')
+    '---string--'
+    >>> to_kebabcase('ABCde.F.G..H..IH')
+    'a-b-cde.f.g..h..i-h'
```

Comment:
`-` after caps, inclusive of the beginning of the string?
See: https://www.npmjs.com/package/kebab-case